src/containers/Tenant/Diagnostics/Partitions/Partitions.tsx

```diff
@@ -111,10 +101,14 @@ export const Partitions = ({path, type, nodes, consumers}: PartitionsProps) => {
     );
 
     useEffect(() => {
-        if (consumersToSelect && consumersToSelect.length && !selectedConsumer) {
+        if (
+            consumersToSelect &&
+            consumersToSelect.length &&
+            (!selectedConsumer || (consumers && !consumers.includes(selectedConsumer)))
+        ) {
```

Comment:
this condition become complex, maybe you want to introduce some variables

Reply:
Updated. Is it better now?

Reply:
much better, thanks